server/poll/poll_test.go

```diff
@@ -239,6 +239,79 @@ func TestUpdateVote(t *testing.T) {
 	}
 }
 
+func TestGetVotedAnswer(t *testing.T) {
+	for name, test := range map[string]struct {
+		Poll     poll.Poll
+		UserID   string
+		Error    bool
+		Expected *poll.VotedAnswerResponse
+	}{
+		"Voted an Answer": {
+			Poll: poll.Poll{
+				ID: testutils.GetPollID(),
+				AnswerOptions: []*poll.AnswerOption{
+					{Answer: "Answer 1", Voter: []string{"a"}},
+					{Answer: "Answer 2", Voter: []string{"b"}},
+					{Answer: "Answer 3", Voter: []string{"b"}},
+				},
+			},
+			UserID:   "a",
+			Error:    false,
+			Expected: &poll.VotedAnswerResponse{PollID: testutils.GetPollID(), UserID: "a", VotedAnswers: []string{"Answer 1"}},
+		},
+		"Voted two Answers": {
+			Poll: poll.Poll{
+				ID: testutils.GetPollID(),
+				AnswerOptions: []*poll.AnswerOption{
+					{Answer: "Answer 1", Voter: []string{"a"}},
+					{Answer: "Answer 2", Voter: []string{"b"}},
+					{Answer: "Answer 3", Voter: []string{"b"}},
+				},
+			},
+			UserID:   "b",
+			Error:    false,
+			Expected: &poll.VotedAnswerResponse{PollID: testutils.GetPollID(), UserID: "b", VotedAnswers: []string{"Answer 2", "Answer 3"}},
+		},
+		"Voted no Answers": {
+			Poll: poll.Poll{
+				ID: testutils.GetPollID(),
+				AnswerOptions: []*poll.AnswerOption{
+					{Answer: "Answer 1", Voter: []string{"a"}},
+					{Answer: "Answer 2", Voter: []string{"b"}},
+					{Answer: "Answer 3", Voter: []string{"b"}},
+				},
+			},
+			UserID:   "c",
+			Error:    false,
+			Expected: &poll.VotedAnswerResponse{PollID: testutils.GetPollID(), UserID: "c", VotedAnswers: []string{}},
+		},
+		"Invalid userID": {
+			Poll: poll.Poll{
+				ID: testutils.GetPollID(),
+				AnswerOptions: []*poll.AnswerOption{
+					{Answer: "Answer 1", Voter: []string{"a"}},
+					{Answer: "Answer 2", Voter: []string{"b"}},
+					{Answer: "Answer 3", Voter: []string{"b"}},
+				},
+			},
+			UserID: "",
+			Error:  true,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			answers, err := test.Poll.GetVotedAnswer(test.UserID)
+			if test.Error {
+				assert.NotNil(err)
```

Comment:
`answers` should be `nil` in this case.
```suggestion
				assert.NotNil(err)
				assert.Nil(answers)
```